common/models/acl.js

```diff
@@ -455,18 +500,96 @@ module.exports = function(ACL) {
 
+        // resolved is an instance of AccessRequest
         var resolved = self.resolvePermission(effectiveACLs, req);
-        if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
-        }
+        resolved.settleDefaultPermission();
         debug('---Resolved---');
         resolved.debug();
+
+        // set authorizedRoles in remotingContext options argument if
+        // resolved AccessRequest permission is ALLOW, else set it to empty object
+        authorizedRoles = resolved.isAllowed() ? authorizedRoles : {};
+        saveAuthorizedRolesToRemotingContext(remotingContext, authorizedRoles);
         return callback(null, resolved);
       });
     });
     return callback.promise;
   };
 
+  function addRoleToAuthorizedRoles(role, authorizedRoles, context) {
+    let valueForAuthorizedRole = ACL.getValueForRoleInAuthorizedRoles;
+    if (typeof valueForAuthorizedRole !== 'function')
+      return;
+    authorizedRoles[role] = valueForAuthorizedRole(context);
+  }
+
+  /**
+   * Get the value to set for and authorized principal in the `authorizedRoles`
+   * object in `remotingContext.args.options` which is set when invoking model
+   * methods via strong-remoting (e.g. REST).
+   *
+   * The default implementation is baremetal and returns true so that the
+   * `authorizedRoles` object looks similar to
+   *
+   * ```js
+   * {
+   *   $authenticated: true,
+   *   $owner: true
+   * }
+   * ```
+   *
+   * Override this method to customize the format of the the `authorizedRoles`
+   * object using the information from the AccessContext object
+   * NOTE: you would require for this to set up a custom ACL model extending the
+   * built-in ACL model, or override the method with a boot script
+   *
+   * Example
+   *
+   * ```js
+   * ACL.getValueForRoleInAuthorizedRoles = function(context) {
+   *   return {
+   *     authorized: true,
+   *     modelName: context.modelName,
+   *     accessType: context.accessType
+   *   };
+   * };
+   * ```
+   *
+   * With this implementation `authorizedRoles` object will look similar to
+   *
+   * ```js
+   * {
+   *   $owner: {
+   *     authorized: true,
+   *     modelName: 'someModel',
+   *     accessType: 'READ'
+   *   }
+   * }
+   * ```
+   *
+   * @options {AccessContext} context An AccessContext instance
+   * @property {Object[]} principals An array of principals.
+   * @property {String|Model} model The model name or model class.
+   * @property {*} modelId The model instance ID.
+   * @property {String} property The property/method/relation name.
+   * @property {String} accessType The access type:
+   *   READ, REPLICATE, WRITE, or EXECUTE.
+   * @returns {Object} The value to set for a role principalId in remotingContext.args.options.authorizedRoles
+   */
+  ACL.getValueForRoleInAuthorizedRoles = function(context) {
+    return true;
+  };
+
```

Comment:
this method is meant to be overriden by user if required. See jsdoc comments
maybe some of this jsdoc should go in the loopback.io doc only?